File: android/src/main/java/com/example/ftdi_serial/SerialDevice.java
package com.example.ftdi_serial;

import java.util.Map;
import java.util.HashMap;

public class SerialDevice {
    private final int vendorId;
    private final int productId;
    private final String deviceName;
    private final String productName;
    private final String manufacturerName;

    public SerialDevice(int vendorId, int productId, String deviceName, String productName, String manufacturerName) {
        this.vendorId = vendorId;
        this.productId = productId;
        this.deviceName = deviceName;
        this.productName = productName;
        this.manufacturerName = manufacturerName;
    }

    public Map<String, Object> toMap() {
        Map<String, Object> result = new HashMap<>();
        result.put("vendorId", vendorId);
        result.put("productId", productId);
        result.put("deviceName", deviceName);
        result.put("productName", productName);
        result.put("manufacturerName", manufacturerName);
        return result;
    }
}

File: lib/serial_device.dart
class SerialDevice {
  // Vendor Id
  final int vendorId;

  // Product Id
  final int productId;
  final String deviceName;
  final String productName;
  final String manufacturerName;

  SerialDevice({
    required this.vendorId,
    required this.productId,
    required this.deviceName,
    required this.productName,
    required this.manufacturerName,
  });

  factory SerialDevice.fromMap(Map<String, dynamic> map) {
    return SerialDevice(
      vendorId: map['vendorId'] as int,
      productId: map['productId'] as int,
      deviceName: map['deviceName'] as String,
      productName: map['productName'] as String,
      manufacturerName: map['manufacturerName'] as String,
    );
  }
}
